test/metabase/models/user_test.clj

```diff
@@ -534,3 +534,22 @@
       (t2.with-temp/with-temp [User {user-id :id} {}]
         (mw.session/with-current-user user-id
           (is (= "v0.47.1" (setting/get :last-acknowledged-version))))))))
+
+(deftest common-name-test
+  (testing "common_name should be present depending on what is selected"
+    (mt/with-temp [User user {:first_name "John"
+                              :last_name  "Smith"
+                              :email      "john.smith@gmail.com"}]
+      (is (= "John Smith"
+             (:common_name (t2/select-one [User :first_name :last_name] (:id user)))))
+      (is (= "John Smith"
+             (:common_name (t2/select-one User (:id user)))))
+      (is (nil? (:common_name (t2/select-one [User :first_name :email] (:id user)))))
+      (is (nil? (:common_name (t2/select-one [User :email] (:id user)))))))
```

Comment:
could you explain why don't we return the common-name in this case?

Reply:
The common_name is a derived property of a User that depends on first_name, last_name, and email columns. Now if you select only email with `(t2/select [User :email] ...)`, for example, then the `define-after-select` function will be called with a transient row with only the email column in it, and will not have enough information to correctly calculate what the common_name of the user is.

Reply:
I've attempted to explain this concisely in the docstring of the add-column-name function.